### PR TITLE
Python : getting started sample

### PR DESCRIPTION
**Pull request by olivakar**
## Purpose
The first sample for python.
Folder structure was changed as imports were causing an issue.

Some of the Paho Client Wrapper code is still unused. Hopefully as we write other samples I will have a better idea what functions are gonna be used.

Some codes are still commented 
9like the certificates path) as the more complex samples need to be tested against that code path. Did not want to uncomment as I am not 100 percent sure.

## Checklist
- [ ] I have read the [contribution guidelines](../CONTRIBUTING.md).
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `main` branch. 
  - [ ] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I will squashed my changes into one with a clear description of the change when I complete the PR.

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  ...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by vishnureddy17** [comment]
I think this sample also needs to subscribe to the `sample/+` topic to align with the [dotnet sample](https://github.com/Azure-Samples/MqttApplicationSamples/blob/4a9ba04c8fc91dafe89dd6f5b06bd0ae56c185f4/scenarios/getting_started/dotnet/getting_started/Program.cs#LL19C47-L19C55).

File: mqttclients/python/pahoclientwrapper/paho_client_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+            properties: Any = None,
+    ) -> None:
+        """
+        event handler for Paho on_subscribe events.  Do not call directly.
+        """
+        # In Paho thread.  Save what we need and return.
+        logger.info(
+            "Received SUBACK for mid {}, granted_qos {}".format(mid, granted_qos)
+        )
+        granted_qos_list = list(granted_qos)
```
Comment (cartertinney) [inline mqttclients/python/pahoclientwrapper/paho_client_wrapper.py:162]: I'm guessing this is an MQTT5 thing I don't fully understand? Why are there (potentially) multiple QoS values?

**Review by cartertinney — APPROVED** [review]

**Review by rido-min — CHANGES_REQUESTED** [review]
The sample fails if the .env file does not have `MQTT_KEY_FILE_PASSWORD`

```
  File "/home/codespace/.python/current/lib/python3.10/site-packages/pahoclientwrapper/paho_client_wrapper.py", line 65, in __init__
    password=connection_settings['MQTT_KEY_FILE_PASSWORD']
KeyError: 'MQTT_KEY_FILE_PASSWORD'
```

File: mqttclients/python/pahoclientwrapper/mqtt_helpers.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+# license information.
+import threading
+import logging
+from typing import Dict, List, Callable, Any, Generic, TypeVar, Optional
+logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+class IncomingAckList(Generic[T]):
```
[the remark this answers is not in the document]
Comment (olivakar) [reply mqttclients/python/pahoclientwrapper/mqtt_helpers.py:12]: discussion needed

File: mqttclients/python/pahoclientwrapper/connection_settings.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    envvar_values = {k: v for k, v in os.environ.items() if k in mqtt_setting_names}
+    default_values = {
+        'MQTT_TCP_PORT': '8883',
+        'MQTT_USE_TLS': 'true',
+        'MQTT_CLEAN_SESSION': 'true',
+        'MQTT_KEEP_ALIVE_IN_SECONDS': '30',
+        'MQTT_CLIENT_ID': ''
+    }
+    
+    final_values = {**default_values, **envvar_values, **envfile_values}
```
[the remark this answers is not in the document]
Comment (olivakar) [reply mqttclients/python/pahoclientwrapper/connection_settings.py:83]: order of precedence is envfile, envvar, and default

**Pull request closed without merging**